src/run_tests.rs

```diff
@@ -33,6 +33,7 @@ impl TestResult {
     /// stay cached unless they pass.
     pub fn met_test_expectations(&self, test: &Test) -> bool {
         match self {
+            TestResult::CompileFailed(_) if !test.compile_fail => true,
             TestResult::Successful(_) if !test.should_panic => true,
```

Comment:
I guess `test.compile_fail` needs to be taken into account here too? So something like

```rust
TestResult::Successful(_) if !test.should_panic => !test.compile_fail,
```

That way a test is marked as failing if it unexpectedly starts to compile (and also runs). The condition could probably be added to the pattern guard instead.

Reply:
Good point. I'll make a small PR.